bin/add_frontend_server.rb

```diff
@@ -1,15 +1,60 @@
 require 'aws-sdk'
 require_relative '../deployment'
+require 'net/ssh'
+require 'net/scp'
+require 'io/console'
+require 'json'
+require 'etc'
 
 MAX_WAIT_TIME = 600 #Wait no longer than 10 minutes for instance creation. Typically this takes a couple minutes
 
+def execute_ssh_on_channel(channel, command, exit_error_string)
+  channel.exec(command) do |ch|
+    ch.on_data do |_, data|
+      $stdout.print data
+    end
+
+    ch.on_extended_data do |_, _, data|
+      $stderr.print data
+    end
+
+    ch.on_request 'exit-status' do |_, data|
+      if data.read_long != 0
+        puts exit_error_string
+        exit(1)
+      end
+    end
+  end
+
+end
+
+options = {}
+
+OptionParser.new do |opts|
+  opts.on('-e', '--environment ENVIRONMENT', 'Environment to add frontend to') do |env|
+    options['environment'] = env
+  end
+
+  opts.on('-h', '--help', 'Print this') { puts options; exit }
+end.parse!
+
+raise OptionParser::MissingArgument, 'Environment is required' if options['environment'].nil?
+
+puts 'Type in your password for gateway.code.org'
+password = STDIN.noecho(&:gets).chomp
+username = Etc.getlogin
+
+Net::SSH.start('gateway.code.org', username, :password => password) do |ssh|
+  puts ssh.exec!('echo \'Verifying connection to gateway\'')
+end
+
 ec2client = Aws::EC2::Client.new
 
 instances = ec2client.describe_instances
 
 #Determine distribution of availability zones, pick the one that has the least capacity among frontend instances
 frontend_instances = instances.reservations.map do |reservation|
-  reservation.instances.select{|instance| instance.tags.detect{|tag| tag.key == 'Name' && tag.value.include?('frontend')}}
+  reservation.instances.select{|instance| instance.state.code == 16 && instance.tags.detect{|tag| tag.key == 'Name' && tag.value.include?('frontend')}}
```

Comment:
`state#name == 'running'` is defined [here](http://docs.aws.amazon.com/sdkforruby/api/Aws/EC2/Types/InstanceState.html#name-instance_method)- I would prefer this to the integer since it is in fact documented in the SDK.